data_linter/validators/pandas_validator.py

```diff
@@ -79,22 +83,230 @@ def read_data(self) -> pd.DataFrame:  # KARIK TODO
-        pass
+
+        meta_cols = [col for col in self.metadata["columns"] if col["name"] in df]
+
+        for i, (_, col) in enumerate(df.iteritems()):
+            self.validate_col(col, meta_cols[i])
+
+    def validate_col(self, col, meta_col):
+
+        self.min_max_test(col, meta_col)
+        self.min_max_length_test(col, meta_col)
+        self.pattern_test(col, meta_col)
+        self.enum_test(col, meta_col)
+        self.nullable_test(col, meta_col)
+        self.datetime_test(col, meta_col)
+
+    def min_max_test(self, col, meta_col):
+        res_dict = _min_max_test(col, meta_col)
+        col_name = meta_col["name"]
+        if res_dict is not None:
+            self.response.add_test_to_col(col_name, "min max numerical", res_dict)
+
+    def min_max_length_test(self, col, meta_col):
+        res_dict = _min_max_length_test(col, meta_col)
+        col_name = meta_col["name"]
+        if res_dict is not None:
+            self.response.add_test_to_col(col_name, "min max length", res_dict)
+
+    def pattern_test(self, col, meta_col):
+        res_dict = _pattern_test(col, meta_col)
+        col_name = meta_col["name"]
+        if res_dict is not None:
+            self.response.add_test_to_col(col_name, "regex pattern", res_dict)
+
+    def enum_test(self, col, meta_col):
+        res_dict = _pattern_test(col, meta_col)
+        col_name = meta_col["name"]
+        if res_dict is not None:
+            self.response.add_test_to_col(col_name, "enum", res_dict)
+
+    def nullable_test(self, col, meta_col):
+        res_dict = _nullable_test(col, meta_col)
+        col_name = meta_col["name"]
+        if res_dict is not None:
+            self.response.add_test_to_col(col_name, "nullable", res_dict)
+
+    def datetime_test(self, col, meta_col):
+        res_dict = _datetime_test(col, meta_col)
+        col_name = meta_col["name"]
+        if res_dict is not None:
+            self.response.add_test_to_col(col_name, "datetime", res_dict)
+
+
+def check_run_validation_for_meta(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        sig = inspect.signature(func)
+        argmap = sig.bind_partial(*args, **kwargs).arguments
+        mc = argmap.get("meta_col")
+        if func.__name__ == "_min_max_test" and _check_meta_has_params(
+            ["minimum", "maximum"], mc
+        ):
+            return func(*args, **kwargs)
+        elif func.__name__ == "_min_max_length_test" and _check_meta_has_params(
+            ["minLength", "maxLength"], mc
+        ):
+            return func(*args, **kwargs)
+        elif func.__name__ == "_pattern_test" and _check_meta_has_params(
+            ["pattern"], mc
+        ):
+            return func(*args, **kwargs)
+        elif func.__name__ == "_enum_test" and _check_meta_has_params(["enum"], mc):
+            return func(*args, **kwargs)
+        elif func.__name__ == "_nullable_test" and _check_meta_has_params(
+            ["nullabe"], mc
```

Comment:
Think we only care about running this test if nullable is False? Also typo `nullabe`